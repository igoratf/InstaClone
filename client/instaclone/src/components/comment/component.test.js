import React from 'react';
import { shallow, mount } from 'enzyme';
import Comment from './comment.component';

describe('Comment component', () => {
  let wrapper;
  let mockProps = {
    author: 'Author',
    content: 'Content'
  };
  const mockSetState = jest.fn();
  jest.mock('react', () => ({
    useState: initial => [initial, mockSetState]
  }));

  beforeEach(() => {
    wrapper = shallow(<Comment {...mockProps} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  })

  it('should render Comment component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render filled like button when like is clicked', () => {
    let mockProps = {
      author: 'Author',
      content: 'Content'
    };

    const newWrapper = mount(<Comment {...mockProps} />);
    newWrapper.find("LikeButton").simulate('click');
    expect(newWrapper.exists("DislikeButton")).toBe(true);
  });

  it('should render like button when liked is clicked twice', () => {
    let mockProps = {
      author: 'Author',
      content: 'Content'
    };

    const newWrapper = mount(<Comment {...mockProps} />);
    newWrapper.find("LikeButton").simulate('click');
    newWrapper.find("DislikeButton").simulate('click');
    expect(newWrapper.exists("LikeButton")).toBe(true);
  });

})