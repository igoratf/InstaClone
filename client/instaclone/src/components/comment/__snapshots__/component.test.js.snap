// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should render Comment component 1`] = `ShallowWrapper {}`;
